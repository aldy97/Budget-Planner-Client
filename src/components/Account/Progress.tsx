import React from "react";
import { Progress, Button } from "antd";
import { MinusOutlined, PlusOutlined } from "@ant-design/icons";

interface ProgressBarProps {
  threshold: number;
  changeThreshold: React.Dispatch<React.SetStateAction<number>>;
}

function ProgressBar({ threshold, changeThreshold }: ProgressBarProps): JSX.Element {
  const handlePlusBtnClick = (): void => {
    changeThreshold(threshold + 10 > 100 ? 100 : threshold + 10);
  };
  const handleMinusBtnClick = (): void => {
    changeThreshold(threshold - 10 < 0 ? 0 : threshold - 10);
  };
  return (
    <>
      <Progress percent={threshold} />
      <Button.Group>
        <Button onClick={handleMinusBtnClick} icon={<MinusOutlined />} />
        <Button onClick={handlePlusBtnClick} icon={<PlusOutlined />} />
      </Button.Group>
    </>
  );
}

export default ProgressBar;
