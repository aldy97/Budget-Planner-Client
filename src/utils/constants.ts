export const COLORS = {
  THEMEBLUE: "#1890ff",
  UNSELECTEDOPTION: "#000000a6",
  ORANGE: "#ffa940",
};

export const ExpenseCategories = [
  "Commute",
  "Meals",
  "Shoppings",
  "Grocery",
  "Veggies",
  "Fruits",
  "Snacks",
  "Sports",
  "Entertainment",
  "Phone Plan",
  "Make up",
  "Housing",
  "Clothing",
  "Children",
  "Elders",
  "Social",
  "Travelling",
  "Wine",
  "Digitals",
  "Car",
  "Medical Care",
  "Books",
  "Learning",
  "Pets",
  "Gifts",
  "Office",
  "Maintainance",
  "Donation",
];

export const IncomeCategories = [
  "Salary",
  "Part-time",
  "Investment",
  "Gifts",
  "Others",
];

export const URL = "http://localhost:7001";
