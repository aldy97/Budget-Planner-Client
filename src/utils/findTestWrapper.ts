import { ReactWrapper } from "enzyme";

export const findTestWrapper = (
  wrapper: ReactWrapper,
  tag: string
): ReactWrapper => {
  return wrapper.find(`[data-test="${tag}"]`);
};
