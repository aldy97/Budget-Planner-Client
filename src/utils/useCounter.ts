import { useEffect, useState } from "react";

export default (): number => {
  const [count, setCount] = useState(0);

  useEffect(() => {
    setTimeout(() => {
      setCount(count + 1);
    }, 1000);
  }, [count]);

  return count;
};
