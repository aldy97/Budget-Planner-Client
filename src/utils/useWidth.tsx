import { useState, useEffect } from "react";

const useInnerWidth = (): number => {
  const [width, setWidth] = useState<number>(window.innerWidth);

  useEffect(() => {
    const handleResize = () => {
      setWidth(window.innerWidth);
    };
    window.addEventListener("resize", handleResize);
  }, [width]);

  return width;
};

export default useInnerWidth;
