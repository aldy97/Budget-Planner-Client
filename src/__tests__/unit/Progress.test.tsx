import React from 'react';
import { mount } from 'enzyme';
import Progress from '../../components/Account/Progress';
import { findTestWrapper } from '../../utils/findTestWrapper';

describe("Progress bar", () => {
    let wrapper: any;
    const fn = jest.fn();
    

    beforeEach(() => {
        wrapper = mount(<Progress threshold = {30} changeThreshold = {fn}></Progress>);
    })

    it("renders without crashing", () => {
        expect(wrapper.length).toBe(1);
        expect(wrapper).toMatchSnapshot();
    })

    it("receives props threshold", () => {
        expect(wrapper.prop("threshold")).toBe(30);
    })

    it("triggers threshold change after button is clicked", () => {
        expect(fn).toHaveBeenCalledTimes(0);
        const buttons = wrapper.find("button");
        expect(buttons.length).toBe(2);
        buttons.at(0).simulate("click");
        expect(fn).toHaveBeenCalledTimes(1);
        buttons.at(1).simulate("click");
        expect(fn).toHaveBeenCalledTimes(2);
    })

    it("the bar renders received threshold", () => {
        const bar = findTestWrapper(wrapper, "bar").at(1);
        expect(bar.find("span").text()).toEqual("30%");
    })
})
