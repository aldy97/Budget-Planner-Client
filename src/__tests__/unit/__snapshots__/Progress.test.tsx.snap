// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress bar renders without crashing 1`] = `ReactWrapper {}`;
